### PR TITLE
ARC 162 CARMA Messenger Core in ROS2 Humble

### PR DESCRIPTION
**Pull request by john-chrosniak**
# PR Details
## Description

This PR contains changes needed to have carma-messenger-core build with ROS2 humble.

## Related GitHub Issue

NA

## Related Jira Key

[ARC-162](https://usdot-carma.atlassian.net/browse/ARC-162)

## Motivation and Context

Towards ROS2 Humble migration

## How Has This Been Tested?

Build tested locally and confirmed to work with CI changes. CARMA Messenger starts up as expected when using the `carma-messenger-config/develop` configuration.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

[ARC-162]: https://usdot-carma.atlassian.net/browse/ARC-162?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: carma-messenger-core/Dockerfile
Comment (MishkaMN) [inline carma-messenger-core/Dockerfile:42]: I think now because this repo only has ROS2, I think it makes sense not to distinguish as install_ros2. Could you rename all those instances of " * _ros2"?

File: carma-messenger-core/Dockerfile
Comment (MishkaMN) [inline carma-messenger-core/Dockerfile:45]: `roslaunch` command should also change to ROS2 version with appropriate launch python files.

Comment (MishkaMN) [reply carma-messenger-core/Dockerfile:45]: 1. That reminds me, ROS1 style `launch` files should be removed. There is one functionality in the `carma-messenger.launch` file that should be converted to ROS2 as well. Not sure if we can do it similarly in ROS2, I assume we can.

```
  <!-- System Alert Publisher to publish simulated DRIVERS_READY status -->
  <node pkg="rostopic" type="rostopic" name="drivers_ready_rostopic_pub"
        args="pub -l /system_alert cav_msgs/SystemAlert '{ type: 5, description: Simulated Drivers Ready }'"
  />
``` 

2. all the `docker-compose.yml` types in `carma-messenger-config` should drop the roscore and ros1_bridge I think because no ROS1 component is running. 
3. these changes require integration testing. 

However, I am also okay holding off on the changes suggested in this particular comment, if your current change is enough to unblock you towards your C1T efforts.

**Review by MishkaMN — CHANGES_REQUESTED** [review]

File: carma-messenger-core/Dockerfile
Comment (john-chrosniak) [reply carma-messenger-core/Dockerfile:45]: Are we planning to migrate bagging to the ROS2 launch file as well?

Comment (MishkaMN) [reply carma-messenger-core/Dockerfile:45]: Yes! We do need the ros2 bag as well although no specific work is setup, so it should done as part of this effort.
Example where we call the ros2 bag here carma-platform https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/carma/launch/carma_src.launch.py#L196

**Review by MishkaMN — APPROVED** [review]
approved.
nitpick: there is some mention about ubuntu 16.04. Let's just remove it. 
https://github.com/usdot-fhwa-stol/carma-messenger/blob/arc-162-ros2-humble/README.md?plain=1#L36

**Pull request merged**
